04-Honduras/02-R/2-data_contruct/02_Construct_standard.R

```diff
@@ -0,0 +1,220 @@
+               DT_CONTRACT_SIGNED, 
+               everything())%>%
+        mutate(publication_start = DT_TENDER_PUB - DT_TENDER_START,
+               submission_time = DT_TENDER_END - DT_TENDER_START,
+               processing_time = DT_CONTRACT_SIGNED - DT_TENDER_START,
+               decision_time   = DT_CONTRACT_SIGNED - DT_TENDER_END)
+      
+      contract_tender <- as.data.table(contract_tender)
+      
+      # check the overlap between contract signature, contract start and end date 
+      check <- contract_tender[,.N, by = "publication_start"] # not consistent 
+      
+      check <- contract_tender[,.N, by = "submission_time"] # consistent: end > start 
+      
+      check <- contract_tender[,.N, by = "processing_time"] # consistent: signature > start 
+      
+      check <- contract_tender[,.N, by = "decision_time"] # consistent: signature > end 
+      
+      
+# Market concentration: number of winners per sector (sector = covid/non-covid) -----
+      
+      supplier_item <- supplier_item%>%
+        mutate(covid_item = ifelse(str_detect(STR_ITEM_DESCRIPTION, "covid")|
+                                     str_detect(STR_ITEM_DESCRIPTION, "COVID")|
+                                     str_detect(STR_ITEM_DESCRIPTION, "Covid")|
+                                     str_detect(STR_ITEM_DESCRIPTION, "coronavirus")|
+                                     str_detect(STR_ITEM_DESCRIPTION, "coronavirus")|
+                                     str_detect(STR_ITEM_DESCRIPTION, "sars-cov-2"),1,0))
+      
+      supplier_item <- as.data.table(supplier_item)
+      
+      market_concentration <- supplier_item[,.N,by = "covid_item"]
+      
+    # Number of different products (UNSPC codes) supplied by the same firm over 1 year (or 6 months). 
+    # [Do firms during covid supply a broader range of different products?]
+    # Besides the shares, explore the number of products sold by the firm before, during and after COVID
+    # Number of different product items (and sectors) sold in one year by the firm
+      
+      firm_item <- supplier_item%>%
+        group_by(ID_PARTY, YEAR)%>%
+        dplyr::summarise(num_unspsc = n_distinct(ID_ITEM_UNSPSC))%>%
+        ungroup()
+      
+      # firms that have increased the product code during COVID
+      firm_uspsc_panel <- firm_item%>%
+        spread(YEAR, num_unspsc)%>%
+        clean_names()%>%
+        select(-x2025, -na, -x2018)%>%
+        replace(is.na(.), 0)%>%
+        mutate(product_increase = case_when(x2020 > x2019 | x2021 > x2019 ~ 1,
+                                            TRUE ~ 0))
+      
+    # Construct and index to classify contracts to define comparison group. To define COVID products: use the TED classification on COVID products
+      covid_item_unspsc <- supplier_item%>%
+        filter(covid_item == 1)%>%
+        select(ID, ID_PARTY, NAME_PARTY, ID_ITEM, ID_ITEM_UNSPSC, STR_ITEM_UNSPSC, STR_ITEM_DESCRIPTION, covid_item)
+      
```

Comment:
Great!!! More variables. It is good to document the code. Can you add explanations on why we decided to look for key words in three levels, why we check for time variables, and so on? Thank you!